src/main/java/org/broadinstitute/hellbender/utils/read/ReadUtils.java

```diff
@@ -1054,6 +1090,43 @@ public static SAMFileWriter createCommonSAMWriterFromFactory(
     }
 
     /**
+     * Create a common SAMFileWriter from a factory for use with GATK tools. Assumes that if the factory has been set
+     * to create an index, the header must be set to coordinate sorted.
+     *
+     * @param outputPath if this file has a .cram extension then a reference is required. Can not be null.
+     * @param referenceFile the reference source to use. Can not be null if a output file has a .cram extension.
+     * @param header header to be used for the output writer
+     * @param preSorted if true then records must already be sorted to match the header sort order
+     * @param factory SAMFileWriterFactory factory to use
+     * @param openOptions (optional) NIO options specifying how to open the file
+     * @return SAMFileWriter
+     */
+    public static SAMFileWriter createCommonSAMWriterFromFactory(
+        final SAMFileWriterFactory factory,
+        final Path outputPath,
+        final File referenceFile,
+        final SAMFileHeader header,
+        final boolean preSorted,
+        OpenOption... openOptions)
+    {
+        Utils.nonNull(outputPath);
+        Utils.nonNull(header);
+
+        if (outputPath.toString().endsWith(CramIO.CRAM_FILE_EXTENSION)) {
+            if (null == referenceFile ) {
+                throw new UserException("A reference file is required for writing CRAM files");
+            }
+            // htsjdk doesn't currently support presorted & outputStream at the same time for CRAM :-(
+            return factory.makeCRAMWriter(header, outputPath, referenceFile);
+        } else if (outputPath.toString().endsWith(".sam")) {
+            return factory.makeSAMWriter(header, preSorted, outputPath);
+        } else {
+            // default to BAM, matching the logic in htsjdk
+            return factory.makeBAMWriter(header, preSorted, outputPath);
+        }
+    }
```

Comment:
I think this should just delegate to makeWriter like the File overload above does. That should do the right thing. Even better, can we consolidate the two methods so the original File one delegates to the this one. That would help with ensuring they're the same (i.e., we should preserve the existing behavior of cloning the header before passing it off to the writer).

Reply:
OK let's give this a try.